src/object_store.go

```diff
@@ -0,0 +1,215 @@
+	if err != nil {
+		return fmt.Errorf("Failed to create new object in bucket %v with key %v: %v", bucket, key, err)
+	}
+
+	return nil
+}
+
+// ObjectExists does Get operation and validates result or error to find out if object exists
+func (o *ObjectStore) ObjectExists(bucket, key string) (bool, error) {
+	log := o.log.WithFields(logrus.Fields{
+		"bucket": bucket,
+		"key":    key,
+	})
+	log.Infof("ObjectExists")
+
+	result := objects.Get(o.client, bucket, key, nil)
+
+	if result.Err != nil {
+		if result.Err.Error() == "Resource not found" {
+			log.Infof("Key %v in bucket %v doesn't exist yet.", key, bucket)
+			return false, nil
+		}
+		return false, fmt.Errorf("Cannot Get key %v in bucket %v: %v", key, bucket, result.Err)
+	}
+
+	return true, result.Err
+}
+
+// ListCommonPrefixes returns list of objects in bucket, that match specified prefix
+func (o *ObjectStore) ListCommonPrefixes(bucket, prefix, delimiter string) ([]string, error) {
+	log := o.log.WithFields(logrus.Fields{
+		"bucket":    bucket,
+		"delimiter": delimiter,
+		"prefix":    prefix,
+	})
+	log.Infof("ListCommonPrefixes")
+
+	opts := objects.ListOpts{
+		Prefix:    prefix,
+		Delimiter: delimiter,
+		Full:      true,
+	}
+
+	allPages, err := objects.List(o.client, bucket, opts).AllPages()
+	if err != nil {
+		return nil, fmt.Errorf("Failed to list objects in bucket %v: %v", bucket, err)
+	}
+
+	allObjects, err := objects.ExtractInfo(allPages)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to extract info from objects in bucket %v: %v", bucket, err)
+	}
+
+	var objNames []string
+	for _, object := range allObjects {
+		objNames = append(objNames, object.Subdir+object.Name)
+	}
+
+	return objNames, nil
+}
+
+// ListObjects lists objects with prefix in all containers
+func (o *ObjectStore) ListObjects(bucket, prefix string) ([]string, error) {
+	log := o.log.WithFields(logrus.Fields{
+		"bucket": bucket,
+		"prefix": prefix,
+	})
+	log.Infof("ListObjects")
+
+	objects, err := o.ListCommonPrefixes(bucket, prefix, "/")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to list objects from bucket %v with prefix %v: %v", bucket, prefix, err)
+	}
+
+	return objects, nil
+}
+
+// DeleteObject deletes object specified by key from bucket
+func (o *ObjectStore) DeleteObject(bucket, key string) error {
+	log := o.log.WithFields(logrus.Fields{
+		"bucket": bucket,
+		"key":    key,
+	})
+	log.Infof("DeleteObject")
+
+	_, err := objects.Delete(o.client, bucket, key, nil).Extract()
+	if err != nil {
+		return fmt.Errorf("Failed to delete object with key %v in bucket %v: %v", key, bucket, err)
+	}
+
+	return err
+}
+
+// CreateSignedURL creates temporary URL for key in bucket
+func (o *ObjectStore) CreateSignedURL(bucket, key string, ttl time.Duration) (string, error) {
+	log := o.log.WithFields(logrus.Fields{
+		"bucket": bucket,
+		"key":    key,
+	})
+	log.Infof("CreateSignedURL")
+
+	return objects.CreateTempURL(o.client, bucket, key, objects.CreateTempURLOpts{
```

Comment:
you are wrapping errors in every other function. Don't you want to do that also in here?

Reply:
Sure I want. Thx for noticing.

Reply:
Added in https://github.com/Lirt/velero-plugin-swift/pull/1/commits/8e08b45d9f9b21218fad32f84d3b67bb9161b262